### PR TITLE
Specify the order we handle repricing

### PR DESCRIPTION
**Pull request by peterbarrow**
Updated the spec so that we process the pegged orders to be repriced as a block and not one by one.

### CODE REVIEW

**Comment by tamlyn10** [comment]
@davidsiska-vega @tamlyn10 @barnabee to review. This has been implemented already.

File: specs/0037-pegged-orders.md [hunk 1]
```diff
@@ -32,7 +32,9 @@ Pegged orders are limit orders where the price is specified of the form `REFEREN
...
+Whenever the reference price changes all the pegged orders that rely on it need to be repriced. First we remove all active pegged orders from the orderbook, then we reprice them to calculate their new price. If we are unable to reprice the order it is parked. We then process each unparked order in the same time sorted way as they were received for margin and if successful we reinsert the order back into the orderbook. Pegged orders which reference a price that has not changed are untouched.
```
Comment (tamlyn10) [inline specs/0037-pegged-orders.md]: To clarify, by this: _in the same time sorted way as they were received_

do you mean in price-time priority (i.e. received into Vega)?

Comment (peterbarrow) [reply specs/0037-pegged-orders.md]: We process pegged orders in the same time order they are submitted. We don't take price into account when processing repricing.

Comment (tamlyn10) [reply specs/0037-pegged-orders.md]: OK, was just checking that the relative time priority between pegs is maintained at each price level?

**Comment by tamlyn10** [comment]
After discussion at spec meeting @tamlyn10 will edit the spec to include relative time priority and @3jtechtest will add to tests!

**Comment by tamlyn10** [comment]
@tamlyn10 to relook at this...

**Comment by edd** [comment]
@tamlyn10 this is still waiting for you

**Comment by davidsiska-vega** [comment]
@tamlyn10 this is still waiting for you

**Review by davidsiska-vega — APPROVED** [review]

**Comment by davidsiska-vega** [comment]
@peterbarrow will merge this but FYI made a small edit that followed (hopefully) what you said in the answer to Tamlyn.

**Pull request merged**